tests/distributed/test_partition.py

```diff
@@ -81,98 +88,220 @@ def create_random_hetero():
+):
+    edata = part.edge_attributes if use_graphbolt else part.edata
+    etype = part.type_per_edge if use_graphbolt else edata[dgl.ETYPE]
+    eid = th.arange(len(edata[dgl.EID]))
+    etype_arr = F.gather_row(etype, eid)
+    eid_arr = F.gather_row(edata[dgl.EID], eid)
+    for etype in g.canonical_etypes:
+        etype_id = g.get_etype_id(etype)
+        eids[etype].append(F.boolean_mask(eid_arr, etype_arr == etype_id))
+        # Make sure edge Ids fall into a range.
+        if store_inner_edge:
+            inner_edge_mask = _get_inner_edge_mask(
+                part, etype_id, use_graphbolt=use_graphbolt
+            )
             inner_eids = np.sort(
-                F.asnumpy(F.boolean_mask(part.edata[dgl.EID], inner_edge_mask))
+                F.asnumpy(F.boolean_mask(edata[dgl.EID], inner_edge_mask))
             )
             assert np.all(
                 inner_eids == np.arange(inner_eids[0], inner_eids[-1] + 1)
             )
 
-        for ntype in g.ntypes:
-            ntype_id = g.get_ntype_id(ntype)
-            # Make sure inner nodes have Ids fall into a range.
-            inner_node_mask = _get_inner_node_mask(part, ntype_id)
-            inner_nids = F.boolean_mask(part.ndata[dgl.NID], inner_node_mask)
-            assert np.all(
-                F.asnumpy(
-                    inner_nids
-                    == F.arange(
-                        F.as_scalar(inner_nids[0]),
-                        F.as_scalar(inner_nids[-1]) + 1,
-                    )
+
+def _verify_node_id_range_hetero(g, part, nids):
+    """
+    check list:
+        make sure inner nodes have Ids fall into a range.
+    """
+    for ntype in g.ntypes:
+        ntype_id = g.get_ntype_id(ntype)
+        # Make sure inner nodes have Ids fall into a range.
+        inner_node_mask = _get_inner_node_mask(
+            part, ntype_id, use_graphbolt=True
+        )
+        inner_nids = F.boolean_mask(
+            part.node_attributes[dgl.NID], inner_node_mask
+        )
+        assert np.all(
+            F.asnumpy(
+                inner_nids
+                == F.arange(
+                    F.as_scalar(inner_nids[0]),
+                    F.as_scalar(inner_nids[-1]) + 1,
                 )
             )
-            nids[ntype].append(inner_nids)
-
-    for ntype in nids:
-        nids_type = F.cat(nids[ntype], 0)
-        uniq_ids = F.unique(nids_type)
-        # We should get all nodes.
-        assert len(uniq_ids) == g.num_nodes(ntype)
-    for etype in eids:
-        eids_type = F.cat(eids[etype], 0)
-        uniq_ids = F.unique(eids_type)
-        assert len(uniq_ids) == g.num_edges(etype)
-    # TODO(zhengda) this doesn't check 'part_id'
-
-
-def verify_graph_feats(
-    g, gpb, part, node_feats, edge_feats, orig_nids, orig_eids
+        )
+        nids[ntype].append(inner_nids)
+
+
+def _verify_node_edge_included_hetero(
+    g, nids, eids, store_inner_edge, store_inner_node
+):
+    if store_inner_node:
```

Comment:
this body looks weird. How is the `store_inner_node/edge` flag related to the check here? In intuition, this flag is about `partition.ndadta['inner_node/edge']. but it's not directly used here.